pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,499 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import ReactDOM from 'react-dom';
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+const TaskState = Object.freeze({
+    READY: 0,
+    RUNNING: 1,
+    COMPLETED: 2,
+    ERROR: 3,
+    CANCELED: 4,
+});
+
+const PromptType = Object.freeze({
+    ASK: 0,
+    ASK_YES_NO: 1,
+    ASK_YES_NO_YESFOREVER: 2,
+    ASK_YES_NO_SAVE: 3,
+    ASK_PASSWORD: 4,
+});
+
+const ProblemState = Object.freeze({
+    REPORTABLE: 0,
+    REPORTING: 1,
+    REPORTED: 2,
+    UNREPORTABLE: 3,
+});
+
+const client = cockpit.dbus("org.freedesktop.problems", { superuser: "try" });
+var reportd_client;
+
+// For one-off fetches of properties to avoid setting up a cache for everything.
+function get_problem_properties(problem) {
+    function executor(resolve, reject) {
+        client.wait().then(() => resolve(client));
+    }
+
+    return new Promise(executor)
+            .then(() => client.call(problem.path,
+                                    "org.freedesktop.DBus.Properties",
+                                    "GetAll", ["org.freedesktop.Problems2.Entry"]));
+}
+
+class FAFWorkflowRow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            problemState: ProblemState.REPORTABLE,
+            process: null,
+            reportLinks: [],
+            message: "",
+        };
+
+        this._onCancelButtonClick = this._onCancelButtonClick.bind(this);
+        this._onReportButtonClick = this._onReportButtonClick.bind(this);
+        this.updateStatusFromBus = this.updateStatusFromBus.bind(this);
+
+        this.updateStatusFromBus();
+    }
+
+    updateStatusFromBus() {
+        get_problem_properties(this.props.problem)
+                .catch(exception => console.error(cockpit.format("Getting properties for problem $0 failed: $1", this.props.problem.path, exception)))
```

Comment:
Shouldn't this also set the state to UNREPORTABLE?

Reply:
Yeah, I think that would make sense.